flutter/lib/src/default_integrations.dart

```diff
@@ -185,10 +185,19 @@ class _LoadContextsIntegrationEventProcessor extends EventProcessor {
         contexts.forEach(
           (key, dynamic value) {
             if (value != null) {
+              final currentValue = eventContexts[key];
               if (key == SentryRuntime.listType) {
                 contexts.runtimes.forEach(eventContexts.addRuntime);
-              } else if (eventContexts[key] == null) {
+              } else if (currentValue == null) {
                 eventContexts[key] = value;
+              } else {
+                if (key == SentryOperatingSystem.type &&
+                    currentValue is SentryOperatingSystem &&
+                    value is SentryOperatingSystem) {
+                  final osMap = {...value.toJson(), ...currentValue.toJson()};
+                  final os = SentryOperatingSystem.fromJson(osMap);
+                  eventContexts[key] = os;
```

Comment:
It's a quick workaround for now, the whole process of merging contexts from the Native and Dart layer should be done differently.

Reply:
What's the issue?

Reply:
https://github.com/getsentry/sentry-dart/issues/942
Flutter sets the `theme` by default, so when merging the contexts, it does not take the `os` from the iOS layer since we don't merge field per field recursively, but rather if the `key` exists or not.

Reply:
Ahh, I see. Yeah, that's a stupid behavior.